dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -28,5 +29,25 @@ def workshop_survey_report
         end
       end
     end
+
+    def teachercon_survey_report
+      unless @workshop.teachercon?
+        raise 'Only call this route for teachercons'
+      end
+
+      survey_report = Hash.new
+
+      survey_report[:this_teachercon] = summarize_workshop_surveys(@workshop.survey_responses, Pd::TeacherconSurvey.options)
+      survey_report[:all_my_teachercons] = summarize_workshop_surveys(
+        Pd::Workshop.where(subject: @workshop.subject).facilitated_by(current_user).flat_map(&:survey_responses),
```

Comment:
note that teachercon workshops are one of two subjects, meaning that if one user were to facilitate both CSP and CSD teachercon workshops, only one type would show up in this response.